### PR TITLE
support map styles for GMapPlot

### LINKED ISSUE #2354: Add support for `styles` in google map `map_options`
To add a lot of flexibility to the presentation of google maps, it would be nice to support the [styles](https://developers.google.com/maps/documentation/javascript/reference) property in map_options, which is simply some style json (see examples [here](https://snazzymaps.com/explore).  

I may be wrong but this could be a simple fix that checks [here](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/common/gmap_plot.coffee#L70) to see if anything is set for `styles` and if so, make it part of the map options.  I don't know how or if you'd expose it at this point in the python api, but it'd be nice for properties like this to be able to be passed on to the map call.

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #2354

### CODE REVIEW

**Comment by bryevdv** [comment]
Screenshot:

screen shot 2015-06-09 at 2 59 31 pm

File: bokehjs/src/coffee/common/gmap_plot.coffee [hunk 1]
```diff
@@ -92,7 +95,7 @@ class GMapPlotView extends Plot.View
         _.each(window._bokeh_gmap_loads, _.defer)
       script = document.createElement('script')
       script.type = 'text/javascript'
-      script.src = "https://maps.googleapis.com/maps/api/js?v=3.exp&
+      script.src = "https://maps.googleapis.com/maps/api/js?v=3&
```
Comment (bryevdv) [inline bokehjs/src/coffee/common/gmap_plot.coffee]: use release version, not experimental version

File: bokeh/properties.py [hunk 2]
```diff
@@ -483,6 +483,21 @@ def validate(self, value):
...
+    JSON.
+    """
+    def validate(self, value):
+        super(JSON, self).validate(value)
+
+        if value is None: return
+
+        try:
+            import json
+            temp = json.loads(value)
```
Comment (damianavila) [inline bokeh/properties.py]: `bokeh/properties.py:497:13: F841 local variable 'temp' is assigned to but never used`

**Comment by damianavila** [comment]
Just a minor flake thing, after fixing that and the not experimental version, I think is ready to go...

**Comment by bryevdv** [comment]
Oh, I fixed the not experimental version, the comment was just to explain that was what the change was.

**Comment by damianavila** [comment]
>Oh, I fixed the not experimental version, the comment was just to explain that was what the change was.

Yes, I looked at the diff on the other way... je je... (trying a new visual diff tool with different colors)

**Comment by bryevdv** [comment]
The only question in my mind is about the `JSON` property. It ends up as a JSON string on the js side (so, requiring a JSON.decode)  I suppose it would be possible to produce an actual js hash automatically, but I am not quite sure how, or whether that would actually be better.

**Comment by damianavila** [comment]
>The only question in my mind is about the JSON property. It ends up as a JSON string on the js side (so, requiring a JSON.decode)

You mean JSON.parse... there is no JSON.decode or maybe I missing it...

> I suppose it would be possible to produce an actual js hash automatically, but I am not quite sure how, or whether that would actually be better.

I am not sure that would actually be better... I think is OK as is (maybe some little comment in the JS side about why you need to use `parse`).

**Comment by bryevdv** [comment]
yes, `parse` was typing from memory. I'll add a comment to the docsrting.

**Comment by damianavila** [comment]
OK, LGTM and it is green, let's merge it.

**Pull request merged**